core/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -3681,4 +3681,58 @@ protected HumanReadableBytes.UnitSystem getUnitSystem()
       return HumanReadableBytes.UnitSystem.DECIMAL;
     }
   }
+
+
+  class DateExpandFunction implements Function
+  {
+    @Override
+    public String name()
+    {
+      return "date_expand";
+    }
+
+    @Override
+    public void validateArguments(List<Expr> args)
+    {
+      validationHelperCheckArgumentCount(args, 3);
```

Comment:
some javadocs on what the three parameters indicate. There was also a plan to introduce granularity like hr,min or sec, are we doing it ?

Reply:
Sure, I will add javadocs. 

> There was also a plan to introduce granularity like hr,min or sec, are we doing it ?

For now, we will go ahead with this signature in Imply.